Assets/Scripts/Layers/Adapters/DataTypeAdapters/NL3DImportAdapter.cs

```diff
@@ -0,0 +1,26 @@
+using Netherlands3D.Twin.Projects;
+using UnityEngine;
+
+namespace Netherlands3D.Twin
+{
+    [CreateAssetMenu(menuName = "Netherlands3D/Adapters/NL3DImportAdapter", fileName = "NL3DImportAdapter", order = 0)]
+    public class NL3DImportAdapter : ScriptableObject, IDataTypeAdapter
+    {
+        public bool Supports(LocalFile localFile)
+        {
+            return localFile.SourceUrl.ToLower().EndsWith(".nl3d");      
+        }
+
+        public void Execute(LocalFile localFile)
+        {
+            var projectDataHandler = FindObjectOfType<ProjectDataHandler>();
```

Comment:
Minor thing: can we cache this?

Reply:
Personaly I dont mind these FindObjectOfType in these scenario's since they only run once and you dont notice the overhead.

Reply:
But yes we can cache this :)

Reply:
Will be a Singleton